File: https:/github.com/lucius-lt/leetcode.git/88-MergeSortedArray/88-MergeSortedArray.c
// Last updated: 7/16/2026, 1:11:51 AM
void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
    int p1 = m - 1, p2 = n - 1, p = m + n - 1;

    while (p2 >= 0) {
        if (p1 >= 0 && nums1[p1] > nums2[p2]) {
            nums1[p] = nums1[p1]; 
            p1--;
        } else {
            nums1[p] = nums2[p2];
            p2--;
        }
        p--;
    }
}

File: https:/github.com/lucius-lt/leetcode.git/88-MergeSortedArray/88-MergeSortedArray.java
// Last updated: 7/16/2026, 1:11:44 AM
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] result = new int[m+n];
        int count=0;
        for(int i=0;i<m;i++){
            result[count++]=nums1[i];
           // count++;
        }
        for(int i =0;i<n;i++){
            result[count++]=nums2[i];
           // count++;
        }
        Arrays.sort(result);

        for(int i=0;i<nums1.length;i++){
            nums1[i]=result[i];
        }
    }
}